### ISSUE TITLE
'sys/sendfile.h' file not found on Mac OS X

### ISSUE THREAD

**Issue by mhart** [mortehu/substring-frequencies#2]
Thanks for fixing #1 !

I'm now getting this:

``` console
$ make
  CXX      substrings.o
  CC       libdivsufsort/divsufsort.o
libdivsufsort/divsufsort.c:343:25: warning: implicitly declaring library function 'malloc' with type 'void *(unsigned long)'
  bucket_A = (saidx_t *)malloc(BUCKET_A_SIZE * sizeof(saidx_t));
                        ^
libdivsufsort/divsufsort.c:343:25: note: include the header <stdlib.h> or explicitly provide a declaration for 'malloc'
libdivsufsort/divsufsort.c:354:3: warning: implicit declaration of function 'free' is invalid in C99 [-Wimplicit-function-declaration]
  free(bucket_B);
  ^
2 warnings generated.
  CC       libdivsufsort/sssort.o
  CC       libdivsufsort/trsort.o
  CC       libdivsufsort/utils.o
libdivsufsort/utils.c:104:5: warning: implicit declaration of function 'free' is invalid in C99 [-Wimplicit-function-declaration]
    free(A);
    ^
libdivsufsort/utils.c:129:24: warning: implicitly declaring library function 'malloc' with type 'void *(unsigned long)'
    if((B = (saidx_t *)malloc((size_t)n * sizeof(saidx_t))) == NULL) { return -2; }
                       ^
libdivsufsort/utils.c:129:24: note: include the header <stdlib.h> or explicitly provide a declaration for 'malloc'
2 warnings generated.
  AR       libsubstrings.a
  CXX      main.o
main.cc:167:46: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
        if (*endptr || csf.threshold_percent < 0 || csf.threshold_percent > 100)
                       ~~~~~~~~~~~~~~~~~~~~~ ^ ~
1 warning generated.
  CXXLD    substring-frequencies
  CC       collect.o
collect.c:7:10: fatal error: 'sys/sendfile.h' file not found
#include <sys/sendfile.h>
         ^
1 error generated.
make: *** [collect.o] Error 1
```

Seems that `sys/sendfile.h` is [Linux only](http://stackoverflow.com/questions/8252698/sys-sendfile-h-not-found-gcc).

Not sure if you want to just mark this project as Linux-only? Would be great obviously to have it working natively on Mac, but up to you whether you can be bothered fixing all non-Linux-specific issues :smile:

**Comment by mortehu** [comment]
Fixed as of `eda8bdd`.

**Comment by mhart** [comment]
Excellent, works a treat. Thanks!
